programs/bench.c

```diff
@@ -709,44 +720,60 @@ BMK_return_t BMK_benchMemAdvanced(const void* srcBuffer, size_t srcSize,
     const void ** const srcPtrs = (const void** const)malloc(maxNbBlocks * sizeof(void*));
     size_t* const srcSizes = (size_t* const)malloc(maxNbBlocks * sizeof(size_t));
 
+
     void ** const cPtrs = (void** const)malloc(maxNbBlocks * sizeof(void*));
     size_t* const cSizes = (size_t* const)malloc(maxNbBlocks * sizeof(size_t));
+    size_t* const cCapacities = (size_t* const)malloc(maxNbBlocks * sizeof(size_t));
 
     void ** const resPtrs = (void** const)malloc(maxNbBlocks * sizeof(void*));
     size_t* const resSizes = (size_t* const)malloc(maxNbBlocks * sizeof(size_t));
 
-    const size_t maxCompressedSize = ZSTD_compressBound(srcSize) + (maxNbBlocks * 1024);   /* add some room for safety */
-    void* compressedBuffer = malloc(maxCompressedSize);
-    void* resultBuffer = malloc(srcSize);
     BMK_timedFnState_t* timeStateCompress = BMK_createTimeState(adv->nbSeconds);
     BMK_timedFnState_t* timeStateDecompress = BMK_createTimeState(adv->nbSeconds);
 
-    BMK_return_t results;
-    int allocationincomplete = !compressedBuffer || !resultBuffer ||  
+    void* compressedBuffer;
+    const size_t maxCompressedSize = dstCapacity ? dstCapacity : ZSTD_compressBound(srcSize) + (maxNbBlocks * 1024);
+    void* resultBuffer = malloc(srcSize);
+
+
+    BMK_return_t results = { { 0, 0, 0, 0 }, 0 };
+    int allocationincomplete;
+
+    if(!dstCapacity) {
+        compressedBuffer = malloc(maxCompressedSize);
+    } else {
+        compressedBuffer = dstBuffer;
+    }
+
+    allocationincomplete = !compressedBuffer || !resultBuffer ||  
```

Comment:
`cCapacities` seems not checked for `malloc()` success.

Since allocation success is tested much later, and there are so many buffers,
try some simple rule,
such as keeping variable order identical,
which would help detecting such missing test.